src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -206,24 +260,105 @@ public static LogicalLocation CreateLogicalLocation(LogicalLocationVersionOne v1
             return logicalLocation;
         }
 
-        public static IList<Notification> CreateNotificationsList(IEnumerable<NotificationVersionOne> v1Notifications)
+        internal LogicalLocation CreateLogicalLocation(string fullyQualifiedLogicalName, string parentKey = null, string decoratedName = null)
         {
-            List<Notification> notifications = null;
+            var logicalLocation = new LogicalLocation
+            {
+                DecoratedName = decoratedName,
+                FullyQualifiedName = fullyQualifiedLogicalName,
+                Name = GetLogicalLocationName(fullyQualifiedLogicalName),
+                ParentKey = parentKey
+            };
+
+            return logicalLocation;
+        }
 
-            if (v1Notifications != null)
+        internal string AddLogicalLocation(LogicalLocation logicalLocation)
+        {
+            if (_currentRun.LogicalLocations == null)
             {
-                notifications = new List<Notification>();
+                _currentRun.LogicalLocations = new Dictionary<string, LogicalLocation>();
+            }
+
+            string fullyQualifiedName = logicalLocation.FullyQualifiedName;
+            string logicalLocationKey = logicalLocation.FullyQualifiedName;
+            int disambiguator = 0;
 
-                foreach (NotificationVersionOne notification in v1Notifications)
+            // We need to do the comparison here too
+            while (_currentRun.LogicalLocations.ContainsKey(logicalLocationKey))
+            {
+                LogicalLocation logLoc = _currentRun.LogicalLocations[logicalLocationKey].DeepClone();
+                logLoc.Name = GetLogicalLocationName(fullyQualifiedName);
+                logLoc.FullyQualifiedName = logicalLocation.FullyQualifiedName;
+
+                if (logicalLocation.ValueEquals(logLoc))
                 {
-                    notifications.Add(CreateNotification(notification));
+                    break;
                 }
+
+                logicalLocationKey = logicalLocation.FullyQualifiedName + "-" + disambiguator.ToString(CultureInfo.InvariantCulture);
```

Comment:
I've seen this code before. How about DRYing it out into `Utilities.CreateDisambiguatedName(string baseName, int index)`? Then if we change the naming convention, we only have to change it in one place.

Reply:
Yup

Reply:
BTW, I'd suggest this encapsulation even if the code only occurred once, for the sake of clarity.

Reply:
Done. In SarifTransformerUtilities for now but I'll probably move this somewhere more global as part of the refactoring PR